File: backend/app/schemas/summary.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class RiskLevel(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


class Topic(BaseModel):
    title: str
    content: str
    source_chunk_ids: list[str]


class DecisionShort(BaseModel):
    topic: str
    decision: str
    source_chunk_ids: list[str]


class Risk(BaseModel):
    title: str
    level: RiskLevel
    description: str


class SummaryDetail(BaseModel):
    overview: str
    topics: list[Topic]
    decisions: list[DecisionShort]
    risks: list[Risk]
    open_questions: list[str]


class SummaryRead(BaseModel):
    meeting_id: str
    summary: SummaryDetail
    generated_at: datetime

File: frontend/meetingPilot-frontend/src/types/summary.ts
export enum RiskLevel {
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
}

export interface Topic {
  title: string
  content: string
  source_chunk_ids: string[]
}

export interface DecisionShort {
  topic: string
  decision: string
  source_chunk_ids: string[]
}

export interface Risk {
  title: string
  level: RiskLevel
  description: string
}

export interface SummaryDetail {
  overview: string
  topics: Topic[]
  decisions: DecisionShort[]
  risks: Risk[]
  open_questions: string[]
}

export interface SummaryRead {
  meeting_id: string
  summary: SummaryDetail
  generated_at: string
}
